marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -38,6 +53,13 @@ instance semigroupState :: Semigroup State where
       { holes: s1.holes <> s2.holes
       , negativePayments: s1.negativePayments <> s2.negativePayments
       , negativeDeposits: s1.negativeDeposits <> s2.negativeDeposits
+      , maxTimeout: max s1.maxTimeout s2.maxTimeout
+      , timeoutNotIncreasing: s1.timeoutNotIncreasing <> s2.timeoutNotIncreasing
+      , letBindings: s1.letBindings <> s2.letBindings
+      , uninitializedUse: s1.uninitializedUse <> s2.uninitializedUse
+      , shadowedLet: s1.shadowedLet <> s2.shadowedLet
+      , trueObservation: s1.trueObservation <> s2.trueObservation
+      , falseObservation: s1.falseObservation <> s2.falseObservation
```

Comment:
If you changed `maxTimeout :: Timeout` to `maxTimeout :: Max Timeout`, I think you can newtype derive the monoid and semigroup instances for `State`.


https://pursuit.purescript.org/packages/purescript-orders/4.0.0/docs/Data.Ord.Max#t:Max

Reply:
very nice! unfortunately `BigInteger` is not bounded so there's no monoid.

Reply:
![sad yoda](http://www.riffsy.com/image/fe20958294d5c9d08a8a5d0aa5e7b1de-2.gif)

Reply:
Made my own `newtype MaxTimeout`